bamboo/abort-if-not-pr-or-redeployment.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+source .bamboo_env_vars || true
+if [[ $GIT_PR != true && $RUN_REDEPLOYMENT != true ]]; then
```

Comment:
should this be `or`?

Reply:
This could probably be defined better.  

This script is called to abort in cases where it's neither a PR nor a REDEPLOYMENT job, by the bootstrap-integration-tests script (e.g. build and deploy the test stack), the cleanup *and* running the regular integration tests. 

Basically, it's saying either

1)  if this is *explicitly* a redeploy job, run the bootstrap, integration tests and cleanup.   The redeploy job is solely dependant on RUN_REDEPLOYMENT.   The opposite of that being skip those steps. 

2) If this is a PR to master from the current branch, run the bootstrap, integraiton tests and cleanup.   The redeploy job will run depending on RUN_REDEPLOYMENT.   

3) . If it's neither, skip those steps as we don't need to run integration tests.

Reply:
Probably the reverse with an 'or' would be the better case, but with bash variables I think we probably want  undef to be the same as false, to be the same as <anything not true>